superset/models/tags.py

```diff
@@ -0,0 +1,233 @@
+
+        session.commit()
+
+    @classmethod
+    def after_update(cls, mapper, connection, target):
+        # pylint: disable=unused-argument
+        session = Session(bind=connection)
+
+        # delete current `owner:` tags
+        query = session.query(TaggedObject.id).join(Tag).filter(
+            TaggedObject.object_type == cls.object_type,
+            TaggedObject.object_id == target.id,
+            Tag.type == TagTypes.owner,
+        )
+        ids = [row[0] for row in query]
+        session.query(TaggedObject).filter(
+            TaggedObject.id.in_(ids)).delete(
+                synchronize_session=False)
+
+        # add `owner:` tags
+        for owner_id in cls.get_owners_ids(target):
+            name = 'owner:{0}'.format(owner_id)
+            tag = get_tag(name, session, TagTypes.owner)
+            tagged_object = TaggedObject(
+                tag_id=tag.id,
+                object_id=target.id,
+                object_type=ObjectTypes.chart,
+            )
+            session.add(tagged_object)
+
+        session.commit()
+
+    @classmethod
+    def after_delete(cls, mapper, connection, target):
+        # pylint: disable=unused-argument
+        session = Session(bind=connection)
+
+        # delete row from `tagged_objects`
+        session.query(TaggedObject).filter(
+            TaggedObject.object_type == cls.object_type,
+            TaggedObject.object_id == target.id,
+        ).delete()
+
+        session.commit()
+
+
+class ChartUpdater(ObjectUpdater):
+
+    object_type = 'chart'
+
+    @classmethod
+    def get_owners_ids(cls, target):
+        return [owner.id for owner in target.owners]
+
+
+class DashboardUpdater(ObjectUpdater):
+
+    object_type = 'dashboard'
+
+    @classmethod
+    def get_owners_ids(cls, target):
+        return [owner.id for owner in target.owners]
+
+
+class QueryUpdater(ObjectUpdater):
+
+    object_type = 'query'
+
+    @classmethod
+    def get_owners_ids(cls, target):
+        return [target.user_id]
+
+
+class FavStarUpdater(object):
+
+    @classmethod
+    def after_insert(cls, mapper, connection, target):
+        # pylint: disable=unused-argument
+        session = Session(bind=connection)
+        name = 'favorited_by:{0}'.format(target.user_id)
+        tag = get_tag(name, session, TagTypes.favorited_by)
+        tagged_object = TaggedObject(
+            tag_id=tag.id,
+            object_id=target.obj_id,
+            object_type=get_object_type(target.class_name),
+        )
+        session.add(tagged_object)
+
+        session.commit()
+
+    @classmethod
+    def after_delete(cls, mapper, connection, target):
+        # pylint: disable=unused-argument
+        session = Session(bind=connection)
+        name = 'favorited_by:{0}'.format(target.user_id)
+        query = session.query(TaggedObject.id).join(Tag).filter(
+            TaggedObject.object_id == target.obj_id,
+            Tag.type == TagTypes.favorited_by,
+            Tag.name == name,
+        )
```

Comment:
I think you could just `query.delete()` at the next line here (as long as you change `session.query(TaggedObject.id)` -> `session.query(TaggedObject)`), no need for another query I think.